lib/wos_client.rb

```diff
@@ -62,11 +62,22 @@ def session_id
   # Resets the session_id and the search client
   # @return [nil]
   def session_close
-    auth.globals[:headers]['Cookie'] = "SID=\"#{session_id}\""
-    auth.call(:close_session)
+    begin
+      auth.globals[:headers]['Cookie'] = "SID=\"#{session_id}\""
+      auth.call(:close_session)
+    rescue Savon::SOAPFault => fault
```

Comment:
so you just need to be sure an exception is caught and logged so the session_id can be set to nil?  (because if you don't rescue it, you'll still have an expired session_id)?

Reply:
Yes, without catching the exception, the method was not resetting the `session_id==nil` and because it is memoized, it could not be reset and it was not valid any more.  This change allows it to be reset, but only by way of the courtesy call to close the session.